api/v1alpha1/catalogentry_types.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	kcpv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/apis/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CatalogEntrySpec defines the desired state of CatalogEntry
+type CatalogEntrySpec struct {
+	// The workspace name where the APIExport locates
+	Workspace string `json:"workspace,omitempty"`
+	// The name of the APIExport
+	ExportName string `json:"exportName,omitempty"`
+	// A list of icon data that can be displayed in UI
+	// +optional
+	Icons []Icon `json:"icons,omitempty"`
+	// A human readable message to describe the information regarding the capabilities
+	// and features that the API provides
+	// +optional
+	Description string `json:"description,omitempty"`
+	// The list of permission claims that will be needed to use this API
+	// +optional
+	PermissionClaims []kcpv1alpha1.PermissionClaim `json:"permissionClaims,omitempty"`
+	// The API information that is available to use
+	// +optional
+	LatestResourceSchemas []string `json:"latestResourceSchemas,omitempty"`
```

Comment:
These in spec are strange. Maybe GVRs or GRs in the status would make sense.

Reply:
same for the claims

Reply:
`LatestResourceSchemas` would be shown to the consumer to list the APIs available when they bind with a particular CE. As @dinhxuanvu mentioned that this could actually be filled by the operator itself when its validating whether the API Export exists. Though `spec` may not be the right place, I'm also wondering if `status` is. What if there are `n` APIResourceSchemas available in the APIExport, but the producer wants to expose only a subset of those to the Catalog through CatalogEntry? Is that a valid scenario, since I'm not sure if the permission model allows binding per APIResourceSchema or for the entire APIExport.

Reply:
I tried to keep `LatestResourceSchemas` the same as what we have in `APIExport` so users can reuse that part. However, based on your feedback, it is too leaky. I will update to just use GVR instead.

For the `PermissionClaims`, I assume it is fine to use it as it is.